test/workbox-precaching/node/controllers/test-PrecacheManager.mjs

```diff
@@ -0,0 +1,304 @@
+import {expect} from 'chai';
+import sinon from 'sinon';
+import clearRequire from 'clear-require';
+
+import expectError from '../../../../infra/utils/expectError';
+import generateTestVariants from '../../../../infra/utils/generate-variant-tests'
+
+// This sets Request mock on the Global scope
+import '../../../mocks/Request.mjs';
```

Comment:
How do you feel about adopting https://github.com/pinterest/service-workers/tree/master/packages/service-worker-mock for our Node tests? I'm assuming it works as expected, but if we find any bugs/issues, we could upstream them and the community would benefit.

Reply:
I didn't realise it had all the mocks - will investigate switching over.